Docs: Git Sync permissions (#123011)
grafana/grafana: docs/sources/as-code/observability-as-code/git-sync/permissions-grafana.md

For Git Sync you need to configure permissions at two layers to function correctly:

- At the Grafana level for repository management and resource access, as described in this document.
- At your Git provider level, to protect your repository. Refer to [Repository protection for Git Sync](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/as-code/observability-as-code/git-sync/gitsync-repo-protection) for more information.

## Grafana role-based permissions

Git Sync integrates with the Grafana standard role-based permission model, which has three levels:

1. **Organization-level:** Default permissions of the `Admin`, `Editor`, or `Viewer` role. Refer to [Roles and permissions](ref:roles-and-permissions) for more details.
2. **Folder-level:** They also apply to all dashboards within it. Refer to [Folder permissions](ref:manage-folder-permissions) for more details.
3. **Dashboard-level:** Refer to [Dashboard permissions](ref:manage-dashboard-permissions) for more details.

{{< admonition type="caution" >}}

Dashboard-level permissions override folder-level permissions, which override organization-level roles.

{{< /admonition >}}

The following applies for Git Sync:

> - **Permissions don't sync**: Folder and dashboard permissions are managed in Grafana only and don't sync to Git. You must configure permissions separately in each Grafana instance that uses the repository

Reviewer: is this true for dashboards @frossano-grafana  ?

Reviewer 2: Isn't it true? i.e. changing permissions to a dashboard in Grafana UI is not syncing to the repo, i.e. AFAIK if you have the same resources in two different instance, they can have different permissions 🤔

Reviewer 3: i don't think we save permissions anywhere, as they are not on app platform. Unless you have some other info @MissingRoberto i would say this is correct

`Editors` can access resources based on the folder/dashboard assignments:

- `dashboards:create`, `dashboards:read`, `dashboards:write`, `dashboards:delete` - On folders/dashboards where assigned Folder Editor or Dashboard Editor
- `folders:create`, `folders:read`, `folders:write`, `folders:delete` - On folders where assigned Folder Editor

#### Viewer role

Users with Viewer role have read-only access to Git Sync:

| Permission Category  | Specific Permissions                                             | What This Allows                                                           |
| -------------------- | ---------------------------------------------------------------- | -------------------------------------------------------------------------- |
| **Read-Only Access** | `provisioning.repositories:read`<br>`provisioning.settings:read` | View repository configurations<br>View Git Sync settings (required for UI) |

`Viewers` can access resources based on the folder/dashboard assignments:

- `dashboards:read` - On folders/dashboards where assigned Folder Viewer or Dashboard Viewer
- `folders:read` - On folders where assigned Folder Viewer

### RBAC for dashboards and folders

> Provisioned dashboards and folders use the Grafana standard permission model. If you modify provisioned resources, Git Sync automatically syncs changes to Git based on your dashboard and folder permissions.

Reviewer: ```suggestion
Provisioned dashboards and folders use the Grafana standard permission model. To modify provisioned resources, you will need permissions over your dashboard and folder.
```

It's not about the syn, that you need normal folder and dashboard permissions.
